### PR TITLE
fix: move provider fields under aws and docker

### PR DESCRIPTION
**Pull request by dkoshkin**
Moving the AWS fields under `aws: {}`  and Docker fields under `docker: {}`

### CODE REVIEW

**Comment by dkoshkin** [comment]
How is it passing in my IDE but not in CI 🤔 
image

**Comment by jimmidyson** [comment]
@dkoshkin Are you running the AWS meta mutator tests? This fails locally for me btw.

**Comment by jimmidyson** [comment]
Pushed a fix to meta handler tests.

**Comment by jimmidyson** [comment]
I was thinking what this would look like if we had a single `ClusterConfig` type with all providers present, but we could validate only one is provided. We would then need to construct the variable schema as necessary for the meta mutator. WDYT? Instead of having separate provider specific top level cluster config types.

**Comment by jimmidyson** [comment]
@dkoshkin I implemented my last suggestion in a new commit - let me know what you think. We an always revert if you don't think it's worth doing it like that.

File: api/v1alpha1/docker_clusterconfig_types.go [hunk 1]
```diff
@@ -4,56 +4,15 @@
...
 	GenericClusterConfig `json:",inline"`
 }
 
-// DockerClusterConfigSpec defines the desired state of DockerClusterConfig.
-type DockerClusterConfigSpec struct {
-	// +optional
-	Docker *DockerSpec `json:"docker,omitempty"`
-
-	//+optional
-	CustomImage *OCIImage `json:"customImage,omitempty"`
```
Comment (dkoshkin) [inline api/v1alpha1/docker_clusterconfig_types.go:30]: Wait, I messed up here with the `CustomImage`.

Comment (jimmidyson) [reply api/v1alpha1/docker_clusterconfig_types.go:30]: I tested this before I pushed so it's really interesting that this didn't break anything re usage via patches, etc :sweat_smile:

**Pull request merged**